.github/workflows/format.yml

```diff
@@ -0,0 +1,34 @@
+name: Check Make Format
+# this workflow checks that go files is formated by `make format`
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  check-format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+      - name: Display go version
+        run: go version
+      - uses: actions/cache@v3
+        with:
+          path: ~/go/bin
+          key: ${{ runner.os }}-go-v1.20-misspell-v0.3.4-goimports-reviser-v3.4.5
+      - name: install misspell
+        run: |
+          type misspell ||
```

Comment:
It looks good, but maybe it is better to use cache-hit, like the example: https://github.com/actions/cache#skipping-steps-based-on-cache-hit